### PR TITLE
ENH update to gfortran 7 for osx

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge-admin, please rerender

**Comment by isuruf** [comment]
Why not go to 8?

**Comment by isuruf** [comment]
I think we should build gfortran 9 and use that for the migration. There's no point using an older compiler. cc @mingwandroid

**Comment by beckermr** [comment]
9?! :(

It took me quite a while to get 7 to build properly and the work is done. All that is left is for us to push through the migration. 

I'd really like to get to 7 for osx. This blocks shipping an updated version of a feedstock that I care about which is my main motivation for all of the extra work here. I also think matching versions on linux and osx is probably preferable to an extra migration later when we upgrade everything.

**Comment by isuruf** [comment]
Sure. One question. Is it okay to wait a couple of weeks more for the r-base migration to end? It's going to start soon and I'm not sure having two big migrations at the same time is a good idea.

**Comment by beckermr** [comment]
Thank you! Whew. 

I doubt they touch the same nodes in the graph but I am more than ok to wait a bit! 

I have an associated PR here for the actual migrator here too (https://github.com/regro/cf-scripts/pull/543). I'd be nice to that reviewed and ready to go in the meantime.

**Comment by isuruf** [comment]
@msarahan, @mingwandroid, @jjhelmus, @kalefranz, just an FYI. You might want to upgrade in defaults to make conda-forge and defaults compatible.

**Comment by scopatz** [comment]
LGTM

**Comment by isuruf** [comment]
There are lots of R packages using fortran, and merging this midway is going to cause issues. Let's wait a bit. At this rate, the R migration will be over in a couple of weeks.

**Comment by CJ-Wright** [comment]
LGTM

**Comment by isuruf** [comment]
FYI, there are 260 packages needing rebuild and we'll be incompatible with defaults. @msarahan, any plans to update defaults?

**Comment by beckermr** [comment]
We could ship both versions for now and then deprecate 4 later. It only doubles the build matrix on OS X so is not as bad as it sounds.

**Comment by beckermr** [comment]
Being compatible with defaults was a major plus of the big compiler migration. Not breaking that may be worth the extra CI time.

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -49,8 +49,7 @@ fortran_compiler:              # [unix or win64]
   - flang                      # [win64]
 fortran_compiler_version:      # [unix or win64]
   - 2                          # [(linux64 or osx) and (environ.get('CF_COMPILER_STACK') == 'comp4')]
-  - 4                          # [osx]
```
Comment (isuruf) [inline recipe/conda_build_config.yaml]: Shouldn't this be kept as it is?

Comment (beckermr) [reply recipe/conda_build_config.yaml]: Do we want to build both? IDK if we answered that question.

Comment (beckermr) [reply recipe/conda_build_config.yaml]: We did build both when I merged those test PRs, but I guess we could do this globally.

Comment (beckermr) [reply recipe/conda_build_config.yaml]: OK. I kept this line so we build everything. When defaults changes we can remove at a later date.

**Comment by beckermr** [comment]
@isuruf ready to hit go on the migration?

**Pull request merged**